### PR TITLE
fix #769

### LINKED ISSUE #769: Update All Python Scripts to nbs similar to `nbdev_update_lib` in v1
I am wondering why `nbdev_update` in v2 works differently in v1. In v1, `nbdev_update_lib` will update all modified python scripts, which is very handy. 

However, in v2, to update changes made in the python module, we need to 
1. `cd python_module`
2. `nbdev_update python_file.py`

This is a cumbersome process. Why not just keep the v1 design by just typing the `nbdev_update` in the parent directory?

### PR DESCRIPTION
**Pull request by hamelsmu**
fixes #769 
Also fixes an additional edge case invlolving `__future__`  related to #775 that I did not catch before

### CODE REVIEW

File: nbs/09_API/02_maker.ipynb [hunk 1]
```diff
@@ -495,7 +495,7 @@
     "        f.write(f\"# AUTOGENERATED! DO NOT EDIT! File to edit: {self.dest2nb}.\")\n",
     "        if last_future > 0: write_cells(cells[:last_future], self.hdr, f)\n",
     "        if self.parse: f.write(f\"\\n\\n# %% auto 0\\n__all__ = {all_str}\")\n",
-    "        write_cells(cells[last_future:], self.hdr, f, 1 if last_future>0 else 0)\n",
```
Comment (hamelsmu) [inline nbs/09_API/02_maker.ipynb:498]: This line of code where the `offset` was being set was causing some cell indexes to be wrong, and causing the wrong cell in the notebook to be overwritten upon `nbdev_update`.  This was causing some notebooks to be updated incorrectly when they had a `from __future__ ...`

When I removed this offset, everything now seems to be updating correctly.  However, I don't quite understand why this offset was here in the first place, so perhaps @jph00 this is something you might be able to help me understand?

**Comment by hamelsmu** [comment]
@jph00 this PR is ready to review, but perhaps not ready to merge.  I have made a comment on a line of code whose original purpose I don't understand, even though removing this line of code fixes the issue I was having.  

I suggest that we discuss that line of code more before merging because I have a feeling that understanding that context is important

**Comment by hamelsmu** [comment]
It should also be worth noting how I went about testing this.  

1. Without making any changes, running `nbdev_update` from the root doesn't result in any git diffs at all in any notebooks (except for one newline in a single notebook).
2. Made changes in **every single module** (the .py files) and studied the resulting diffs in the notebook(s) to make sure that the notebooks are updating correctly.  I tried making changes in the last cell an also the first cell and then also middle cells.

**Comment by jph00** [comment]
This is such an awesome PR - really love the testing process and description, and double-checking the odd line of code (which appears to be because I changed how things were exported but never changed the index!)

lvgtm (looks *very* good to me)

**Pull request merged**